### DISCUSSION TITLE
v0.3.0 Release

### DISCUSSION THREAD

**Discussion by HEIGE-PCloud** [HEIGE-PCloud/DoIt#842] [General]
<hr /><em>This discussion was created from the release <a href='https://github.com/HEIGE-PCloud/DoIt/releases/tag/v0.3.0'>v0.3.0 Release</a>.</em>

**Comment by yeahwong** [comment]
这个新版本，使用Safari浏览器，文章的banner在首页会显示图片的原始大小，而不是统一对齐了，但在chrome里面是正常的，请问这个需要额外设置吗？旧版本就没这个问题。

**Reply by HEIGE-PCloud** [reply]
我的确针对首页图片做了一些修改，我移除了对于 [lazysizes](https://github.com/aFarkas/lazysizes) 的依赖。之前 lazysizes 被用于图片懒加载，而由于现在大多数现代浏览器都支持了原生的[懒加载](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#browser_compatibility)所以就把这个额外的 JavaScript 依赖给移除了。

图片的大小问题似乎和 CSS 中 `object-fit` 的属性有关。在上述改动中我也确实[调整了](https://github.com/HEIGE-PCloud/DoIt/commit/03a3b3a6bf528799298b6dc9b3df68d33eec2961)这个属性，但这个调整应该是不会破坏原有的行为的。

Safari 似乎有一些关于 `object-fit` 的问题 https://stackoverflow.com/questions/60192601/object-fit-cover-not-working-correctly-on-safari

但我并不确定这个问题影响哪些版本，我在本地 (Safari 16.3) 也没有复现。
<img width="1453" alt="image" src="https://user-images.githubusercontent.com/52968553/216998309-4ff93494-4ebb-45bd-9b82-c118ce4658bc.png">

**Reply by HEIGE-PCloud** [reply]
https://developer.apple.com/forums/thread/707247

**Reply by yeahwong** [reply]
是不是演示站的这些文章的banner的宽度已经超过了首页文章列表的宽度，所以才是正常显示。如果这些图片的宽度小于列表的宽度，它会显示原始大小。

如果文章的banner比较小，其在文章页面也会显示原始大小。

另外，首页文章列表宽度是不是减小了？感觉没之前的更好啊。

**Reply by yeahwong** [reply]
文章内页，如果banner的大小小于标准宽度，Chrome也会如图这样显示。

<img width="954" alt="image" src="https://user-images.githubusercontent.com/5824590/217023717-ed2a2093-59eb-4a9f-8632-ac909cd74e45.png">

如果是Safari的话，首页会如此。

但之前版本的doit并不是这样的，它会给放大到等宽。

**Reply by HEIGE-PCloud** [reply]
> 文章内页，如果banner的大小小于标准宽度，Chrome也会如图这样显示。
> 
> <img alt="image" width="954" src="https://user-images.githubusercontent.com/5824590/217023717-ed2a2093-59eb-4a9f-8632-ac909cd74e45.png">
> 
> 如果是Safari的话，首页会如此。
> 
> 但之前版本的doit并不是这样的，它会给放大到等宽。

我提交了一个修复，你可以试试看 https://github.com/HEIGE-PCloud/DoIt/commit/fb12becf9da93f60e72e4f895279a77848eeb826

**Reply by yeahwong** [reply]
好啦，非常地谢谢。

**Comment by foxdie987** [comment]
`Error: Error building site: JSBUILD: failed to transform "js/theme.js" (text/javascript): "/opt/buildhome/repo/assets/js/theme.js:4:24": Could not resolve "topbar"`
I get this error when updated to 0.3.0, any idea how to fix it? Thanks.
